### DISCUSSION TITLE
Signal strenth value

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#415] [Q&A] [answered]
Where does the NOAA page (https://dcs1.noaa.gov/Account/FieldTest) get the extra digit in the signal strength from (32.3 instead of 32)?
![image](https://github.com/opendcs/opendcs/assets/140545535/16a3ab65-7c6b-4f6d-845d-b6d2d82f3ad1)

From the [documentation](https://opendcs-env.readthedocs.io/en/7.1/lrgs-userguide.html#dcp-message-content), the header should only have a 2-digit integer as the signal strength. Here is the corresponding message where 32 is the signal strength

A082A406 23348231838 G **_32_** -0 H N 204 W UP 00012 `BST@BB@BBi

**Comment by MikeNeilson** [comment]
That's probably a question for @NathanHolcomb , or at the least he'd likely better know who to ask.

**Reply by NathanHolcomb** [reply]
From my understanding, that level of accuracy is not terribly helpful and due to size limitation in the GOES header other useful information was included instead of the extra digit. 

I've reached out to Bret Betsill, with Microcom (the contractor who has been working on DADDS), to get some more background but I'm fairly sure it is purely a space issue.

**Comment by NathanHolcomb** [comment, marked as answer]
Here is the full answer from Brett Betsill with Mircocom along with some supporting documentation he provided for anyone interested:

The simple answer is that the extra digit comes from the DAMS-NT demodulator.  The more complicated answer is that it comes from the DAMS-NT and is ingested in DADDS via a special protocol called the DAMS-NT HiQ or High Quality socket interface that Microcom developed many years ago.

When we first developed the DAMS-NT system, we had to implement the original DAMS-NT DCP socket protocol that I believe Mike Maloney developed; circa 2001-2003.  The original DAMS-NT DCP protocol was patterned after the old DAPS 37-byte header that you have shown.  However, the system specification for the DAMS-NT DCS receiver required improved measurement capability.

Since we could not deviate from the original DAMS-NT DCP spec, we had to develop the DAMS-NT HiQ interface to be able to demonstrate the improved measurement capability.  Later on, when we developed DADDS, we utilized the DAMS-NT HiQ interface, which is why DADDS has always shown the extra digit in the Signal Strength (Sig) field.  It is also why DADDS shows the other quality measurements of Phase Noises (Phs), % Quality or Good Phase Percentage (Qual), Frequency Deviation (Freq).

Attached is a PDF of a presentation I put together many years ago that details the Message Stats coming from the DAMS-NT demods; slides 16 and 17 compare the old 37-byte DAPS header to the stats that show up on DADDS.

I should also note that while the OpenDCS/LRGS systems at NOAA still utilize the DAMS-NT DCP socket connection, in 2019/2020, Microcom was tasked by NOAA to update the DAMS-NT DCP protocol to includes the higher quality stats.  This became V8.2 of the DAMS-NT Network Interface Specification, which I have also attached. 

Following the DAMS-NT V8.2 update, Mike Maloney updated the DDS protocol to support the higher quality message stats enhancement.  These improved stats are also sent in the updated DCS HRIT stream.

However, as you know, the standard 37-byte DCS header is still being used quite a bit.  While the extra digit in the signal level isn’t terribly important, the improved frequency offset is more useful.  The frequency offset in the legacy 37-byte header only has a resolution of 50 Hz.  While this was ok when the DCPs had a +/-500Hz uncertainty, the new CS2 standard has dropped the frequency uncertainty to +/-150Hz so a 50 Hz resolution is a bit coarse.  The other stats can also be useful for troubleshooting issues with platforms.

[DAMS-NT Interface Specification V8.2.pdf](https://github.com/opendcs/opendcs/files/13707020/DAMS-NT.Interface.Specification.V8.2.pdf)

[DCS_MessageStats201909.pdf](https://github.com/opendcs/opendcs/files/13707024/DCS_MessageStats201909.pdf)
